server/src/room.ts

```diff
@@ -10,26 +11,49 @@ const roomRouter = express.Router();
 // get room
 roomRouter.get("/:roomId", async (req, res) => {
   const { roomId } = req.params as { roomId: string };
+  const address = req.user ? req.user.payload.id : "";
   if (process.env.NODE_ENV !== "production") {
     return res.status(200).send({
       name: "default",
     });
   }
 
   const doc = await collection.doc(roomId).get();
-
   if (!doc.exists) {
     return error(res, "room does not exist");
   }
-
-  res.status(200).send(doc.data());
+  const contractAddress = await doc.get("contractAddress");
+  let visible = true;
+  //If room has contract address set visibility to false until finds a permission.
+  if(contractAddress){
+    visible = false;
+
+    if(address){
+      const tokens = await axios.get('https://api.covalenthq.com/v1/137/address/'+ address +'/balances_v2/?nft=true&key=ckey_d79400f2126140158e8c07f7968');
```

Comment:
dont need to use an api for this. we can query the blockchain for free using a local provider like through ethers. Check server/src/tokens.ts
to see how to setup a provider and contract, except instead of `transfer` we can call `balanceOf`